### PR TITLE
Adjust the position of the adder on window.resize

### PR DESCRIPTION
**Pull request by esanzgar**
First commit is a refactor to use the `ListenerCollection` utility.

Second commit is the actual fix for #3194.

Applying this PR, this is how the `adder` repositions on `window.resize`:

https://user-images.githubusercontent.com/8555781/112584454-2b223500-8df8-11eb-94c1-3d72f65f42bf.mov

### CODE REVIEW

File: src/annotator/test/guest-test.js [hunk 1]
```diff
@@ -487,6 +492,22 @@ describe('Guest', () => {
         assert.notCalled(guest.crossframe.call);
       }
     });
+
+    it('reposition the adder on window "resize" event', () => {
+      sandbox.stub(guest, '_repositionAdder').callThrough();
+      sandbox.stub(guest, '_onSelection');
+      window.dispatchEvent(new Event('resize'));
+
+      // If the adder is not displayed on 'window.resize' event does nothing
```
Comment (LMS007) [inline src/annotator/test/guest-test.js:501]: Maybe reword this a little...

"Does not call _onSelection() if the adder is hidden"

File: src/annotator/test/guest-test.js [hunk 2]
```diff
@@ -487,6 +492,22 @@ describe('Guest', () => {
...
+      assert.called(guest._repositionAdder);
+      assert.notCalled(guest._onSelection);
+
+      // If the adder is displayed then it calls 'guest._onSelection'
```
Comment (LMS007) [inline src/annotator/test/guest-test.js:505]: Maybe reword this a little...

"Calls _onSelection() if the adder is visible"

And I would actually just make this a second test since its testing a different code path.

**Review by LMS007 — CHANGES_REQUESTED** [review]
Code changes look good. I just had some minor suggestions to break the test up into 2 tests.  And there seems to be a missing condition that codecov is barking at. 

This is a noticeable improvement for HTML documents. 👍

File: src/annotator/test/guest-test.js (hunk 1 above)
Comment (esanzgar) [reply src/annotator/test/guest-test.js:501]: Good idea

File: src/annotator/test/guest-test.js (hunk 2 above)
Comment (esanzgar) [reply src/annotator/test/guest-test.js:505]: I have split the tests in two. I name the test more in terms of what it does, more than what function calls.

**Comment by esanzgar** [comment]
On a third commit, I will add tests for some of the lines that are missing code coverage: 
image

**Comment by esanzgar** [comment]
@LMS007, would you mind if we merge this PR and I improve the code coverage in another PR?

I have created #3203 to keep track.

**Review by LMS007 — APPROVED** [review]
Thanks for waiting Eduardo. Yep, just follow up with the coverage additions
